django/db/backends/postgresql/schema.py

```diff
@@ -174,8 +185,41 @@ def _alter_column_type_sql(
         }
         old_is_auto = old_internal_type in auto_field_types
         new_is_auto = new_internal_type in auto_field_types
-        if new_is_auto and not old_is_auto:
+        serial_field_types = {
+            "SerialField",
+            "BigSerialField",
+            "SmallSerialField",
+        }
+        old_is_serial = old_internal_type in serial_field_types
+        new_is_serial = new_internal_type in serial_field_types
+
+        if old_is_auto and new_is_serial:
             column = strip_quotes(new_field.column)
+            self._drop_identity(table, column)
+            sequence = self._create_sequence(table, column)
+            self._reset_sequence(table, column)
+            return (
+                (
+                    self.sql_set_default_sequence
+                    % {
+                        "table": self.quote_name(table),
+                        "column": self.quote_name(column),
+                        "sequence": self.quote_value(sequence),
+                    },
+                    [],
+                ),
+                [],
+            )
```

Comment:
I would pull some of this out into a new variable:

```suggestion
            set_default_sequence = self.sql_set_default_sequence % {
                "table": self.quote_name(table),
                "column": self.quote_name(column),
                "sequence": self.quote_value(sequence),
            }
            return ((set_default_sequence, []), [])
```

And similarly below.

Reply:
Yes, this part looks messy... maybe I could put this in a function, e.g. `def _set_default_sequence_sql`, seems that's what others did in the other schema editor classes.